compiler/src/dotty/tools/dotc/core/NameKinds.scala

```diff
@@ -61,8 +61,7 @@ object NameKinds {
     def infoString: String
   }
 
-  /** The kind of SimpleNames */
-  val SimpleNameKind = new NameKind(UTF8) { self =>
+  object SimpleNameKind extends NameKind(UTF8) { self =>
```

Comment:
I think ultimately, we should just make the name registration thread-safe, it doesn't cost much and it's much easier to reason about than remembering which NameKinds have side-effects, etc.